explorer/websockethandlers.go

```diff
@@ -95,7 +129,47 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 
 				case "getmempooltxs":
 					exp.MempoolData.RLock()
-					msg, err := json.Marshal(exp.MempoolData)
+
```

Comment:
Unfortunately, this defeats the purpose of the lock.  `exp.MempoolData` is a pointer.  So when you read the data via `mempoolData`, it's the same data the lock was trying to protect.

If all the code that is using the data is too slow (e.g. `GetExplorerTx` accesses the DB), you will have to make a deep copy of the mempool data, not just of the pointer.  This includes a deep copy of the transaction slices since copying the mempool data struct does not allocate new underlying buffers for the slice data, they will be shared with the old and new slices.